src/lib/network/edit_nic_name.rb

```diff
@@ -63,12 +63,16 @@ def run
           next
         end
 
+        current_rule = LanItems.GetItemUdevRule(LanItems.current)
+
         if UI.QueryWidget(:udev_type, :CurrentButton) == :mac
           rule_key = MAC_UDEV_ATTR
           rule_value = @mac
+          LanItems.RemoveKeyFromUdevRule(current_rule, "ATTR{dev_port}")
         else
           rule_key = BUSID_UDEV_ATTR
           rule_value = @bus_id
+          LanItems.ReplaceItemUdev("ATTR{dev_port}", "ATTR{dev_port}", dev_port(@old_name))
```

Comment:
so you should think of generalization of ```use_udev_rule_for_bonding!``` e.g. turning it into ```..._for_busid``` or even some more intrusive refactoring of this piece of sh.. errr code. At least you do not check if ```dev_port``` returns something none empty here.